sycl/test-e2e/Basic/accessor/accessor.cpp

```diff
@@ -837,17 +837,37 @@ int main() {
               B(b, cgh);
           auto B1 = b1.template get_access<sycl::access::mode::read_write>(cgh);
 
-          cgh.single_task<class acc_with_zero_sized_buffer>(
+          cgh.single_task<class acc_with_unused_zero_sized_buffer>(
               [=]() { B[0] = 1; });
         });
       }
-      assert(!"invalid device accessor buffer size exception wasn't caught");
-    } catch (const sycl::invalid_object_error &e) {
-      assert(e.get_cl_code() == CL_INVALID_VALUE);
     } catch (sycl::exception e) {
       std::cout << "SYCL exception caught: " << e.what();
-      return 1;
+      assert(false && "Unexpected exception!");
+    }
+  }
+
+  // Used accessor with buffer size 0.
+  {
+    bool CaughtException = false;
+
+    try {
+      sycl::buffer<int, 1> b{0};
+
+      sycl::queue queue;
+      queue.submit([&](sycl::handler &cgh) {
+        auto B = b.template get_access<sycl::access::mode::read_write>(cgh);
+
+        cgh.single_task<class acc_with_zero_sized_buffer>([=]() {
+          if (!B.empty())
+            B[0] = 1;
+        });
```

Comment:
Discussed offline. This should not throw.

Reply:
Thanks @steffenlarsen , the test are changed as advised.